filemutex_flock.go

```diff
@@ -57,3 +58,15 @@ func (m *FileMutex) RUnlock() {
 	}
 	m.mu.RUnlock()
 }
+
+// Close does an Unlock() combined with closing and unlinking the associated
+// lock file. You should create a New() FileMutex for every Lock() attempt if
+// using Close().
+func (m *FileMutex) Close() {
+	if err := syscall.Flock(m.fd, syscall.LOCK_UN); err != nil {
+		panic(err)
+	}
+	syscall.Close(m.fd)
+	syscall.Unlink(m.path)
```

Comment:
I'm a bit worried about a function called "Close" also deleting the file. If I were using this API, I'd expect Close() to have similar semantics to os.File.Close, which closes the file descriptor but does not actually delete the file. How about having `Close()` simply close the file descriptor, and then adding a new function `CloseAndRemove()` to additionally unlink the path?

Reply:
@sb10 As an API user, I would be very surprised that `Close`ing a file handle would deleting the file.

In our use case for CNI plugins, we are [opening the lock on a directory containing data](https://github.com/containernetworking/plugins/blob/7f98c94613021d8b57acfa1a2f0c8d0f6fd7ae5a/plugins/ipam/host-local/backend/disk/backend.go#L50) that needs to outlive 
 multiple cycles of `Lock` and `Close`.

We would not be able to use this library if `Close` caused our data directory to be deleted.